src/src/lms/lms.service.ts

```diff
@@ -419,5 +379,428 @@ export class LMSService {
+						certificateTemplate = certificateTemplate.replace(
+							'{{user_id}}',
+							uid,
+						);
+
+						//qr code
+						try {
+							let qr_code_verify_link =
+								this.configService.get<string>(
+									'LMS_CERTIFICATE_VERIFY_URL',
+								) +
+								'' +
+								certificate_id;
+
+							let modifiedHtml = null;
+							const modified = await new Promise(
+								(resolve, reject) => {
+									qr.toDataURL(
+										qr_code_verify_link,
+										function (err, code) {
+											if (err) {
+												resolve(null);
+												return;
+											}
+
+											if (code) {
+												const newHtml = code;
+
+												const root =
+													parse(certificateTemplate);
+
+												// Find the img tag with id "qrcode"
+												const qrcodeImg =
+													root.querySelector(
+														'#qr_certificate',
+													);
+
+												if (qrcodeImg) {
+													qrcodeImg.setAttribute(
+														'src',
+														newHtml,
+													);
+													modifiedHtml =
+														root.toString();
+
+													resolve(modifiedHtml);
+												} else {
+													resolve(null);
+												}
+											} else {
+												resolve(null);
+											}
+										},
+									);
+								},
+							);
+							if (modifiedHtml != null) {
+								certificateTemplate = modifiedHtml;
+							}
+						} catch (e) {
+							console.log(e);
+						}
+						//update certificate html
+						const lmsCertificate = new LMSCertificateDto({
+							certificate_html: certificateTemplate,
+						});
+						await this.updateCertificateHtml(
+							lmsCertificate,
+							certificate_id,
+						);
+					} else {
+						//error in create certificate
+					}
+				}
+				// Update in attendance data in database
+				await this.markCertificateStatus(
+					userTestData?.id,
+					issue_status,
+				);
+			}
+		}
+	}
+	async CapitalizeEachWord(sentence) {
+		if (sentence == null || sentence === '') {
+			return '';
+		} else {
+			const arr = sentence.split(' ');
+			for (var i = 0; i < arr.length; i++) {
+				arr[i] = arr[i].charAt(0).toUpperCase() + arr[i].slice(1);
+			}
+			const c_sentence = arr.join(' ');
+			return c_sentence;
+		}
+	}
+	async fetchTestTrackingData(limit: number) {
+		const query = `
+			query Getlms_test_tracking {
+				lms_test_tracking(
+				where:{
+					certificate_status:{
+						_is_null: true
+					}
+				},
+				limit: ${limit}
+				){
+					id
+					user_id
+					test_id
+					score
+					context
+					context_id
+				}
+			}
+			`;
+		try {
+			const data_list = (await this.hasuraService.getData({ query }))
+				?.data?.lms_test_tracking;
+			//console.log('data_list cunt------>>>>>', data_list.length);
+			//console.log('data_list------>>>>>', data_list);
+			return data_list;
+		} catch (error) {
+			console.log('fetchTestTrackingData:', error, error.stack);
+			return [];
+		}
+	}
+
+	async issueCertificateHtml(lmsCertificateDto: LMSCertificateDto) {
```

Comment:
give better name like prepateCertificateHtml

Reply:
given name as prepareCertificateHtml and moved to cron folder